evm/src/main/java/org/hyperledger/besu/evm/processor/AbstractMessageProcessor.java

```diff
@@ -208,31 +198,45 @@ public void process(final MessageFrame frame, final OperationTracer operationTra
       codeExecute(frame, operationTracer);
 
       if (frame.getState() == MessageFrame.State.CODE_SUSPENDED) {
-        if (operationTracer != null) {
-          operationTracer.traceContextExit(frame);
-        }
         return;
       }
 
       if (frame.getState() == MessageFrame.State.CODE_SUCCESS) {
+        if (operationTracer != null) {
+          operationTracer.traceContextExit(frame);
+        }
```

Comment:
This is unneeded.  We should only do this on `COMPLETED_SUCCESS` and `COMPLETED_FAILURE`